.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,63 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+
+on:
+    - pull_request
+    - push
+
+name: "Continuous Integration"
+
+jobs:
```

Comment:
I suggest splitting this into multiple workflows, since you probably want to run only some od the steps with dependency permutations

Reply:
Here only one job was coded, which of course is sub-optimal.

But in cases such this one where the matrix and goals are simple, one workflow with multiple jobs should be enough and easier to maintain instead of multiple jobs, wouldn't it?

Reply:
> hi @drupol
> 
> I took a quick look and quickly realised that this PR is definitely too big for me.
> 
> I'd be happy to accept one where you add only the Github Actions part

Ok will do that, do you want to [keep this PR as a reminder for the other things and create a new one with only the CI stuff] or [I force push and discard the other commits] ?

Reply:
I'd be curious to know what the Marco's have in mind :-)

Reply:
My main concern is to make this easy to upgrade by always checking ahead (latest deps), behind (old deps) and locked deps.

Sadly, I didn't get to put time in this today, as I ended up working till late again :(

Reply:
Oooh that, I will push a commit tonight for that :-)

Reply:
Ok I pushed commit https://github.com/marcosh/lamphpda/pull/3/commits/6312dd53897554127a19bd4756c93bcdcee6c53d that implements it.